### PR TITLE
Added JOC testnet and mainnet specification file

### PR DESCRIPTION
**Pull request by shashankshampi**
Fixes Closes Resolves #

_Please choose one of the keywords above to refer to the issue this PR solves followed by the issue number (e.g. Fixes #000). If no issue number, remove the line. Also, remove everything marked optional that is not applicable. Remove this note after reading._

## Added JOC chain spec for mainnet and testnet

- _List the changes_
- Added .cfg file for joc testnet and joc mainnet
- Added chainspec json file for joc mainnet and joc testnet

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [x] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [x] Yes
- [ ] No

In this update, we're excited to announce the successful onboarding of JOC Mainnet and Testnet to leverage the powerful capabilities of the Nethermind client for synchronization. This enhancement ensures smoother operations and improved efficiency for syncing on both networks.

Key Highlights:

Integration of the Nethermind client for JOC Mainnet and Testnet synchronization.
Enhanced performance and reliability during syncing processes.
Streamlined operations for improved user experience.
This update marks a significant step forward in optimizing the syncing functionality, providing users with a more robust and efficient experience on both the JOC Mainnet and Testnet environments.

About JOC:

Ethereum-compatible blockchain infrastructure operated by Japanese companies and ideal for business use. JOC prioritizes legal and technical practicality and provides a safe, secure, and cost-effective use for anyone worldwide to use with confidence.
https://www.japanopenchain.org/en/about

## Remarks

Similar configuration of Geth.

Geth command to sync JOC:
```
JOC Mainnet with Geth:

geth init --datadir data data/genesis.json

geth --datadir data --networkid=81
--bootnodes="enode://d4c1196326527c13cb318fb062571d9ae25393cbaa06222b3e57c
a6407eeac550cf0fd148250282fdcb48e64877f3451d7a8cca281d9a0364c5739462976dfb
5@54.178.193.185:30303" -http.api
"clique,personal,eth,net,web3,miner,admin" --http.corsdomain="*" --http
--http.addr="0.0.0.0" --http.port=8546 --snapshot=false --syncmode="full"
```

```
JOC Testnet with Geth:

geth init --datadir geth_test_data data/testnet_genesis.json

geth --datadir geth_test_data --networkid=361257328 --bootnodes="enode://8aa6f351eff4bee5d3a6a72ca5820fac65274e9dbd63e13d060682a5228000ab960ff8c177d7cf66b0555859b7eabbc866b71625626a11856e3573bf0592bfed@18.183.231.155:30303" -http.api "clique,personal,eth,net,web3,miner,admin" --http.corsdomain="*" --http --http.addr="0.0.0.0" --http.port=8546 --snapshot=false --syncmode="full"
```

JOC official Geth documentation for testnet and mainnet

[Procedure for Connecting to JOC Mainnet.pdf](https://github.com/NethermindEth/nethermind/files/13488861/Procedure.for.Connecting.to.JOC.Mainnet.pdf)
[Procedure for Connecting to JOC Testnet.pdf](https://github.com/NethermindEth/nethermind/files/13488862/Procedure.for.Connecting.to.JOC.Testnet.pdf)

### CODE REVIEW

**Comment by kamilchodola** [comment]
@shashankshampi 
Please also add as a reference here Geth genesis files for both mainnet and testnet and if exists, Geth cofigs for those.

File: src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        "Timeout": 20000,
+        "Host": "127.0.0.1",
+        "Port": 8545,
+        "EngineHost": "127.0.0.1",
+        "EnginePort": 8551,
+        "EngineEnabledModules": "net,eth,subscribe,engine,web3,client,debug,trace,parity"
+    },
+    "Merge": {
+        "Enabled": true,
+        "FinalTotalDifficulty": "1"
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg:30]: are you sure it is PoS network without difficulty increase? I think it is clique and we shouldn't have FinalTotalDifficulty and Merge.Enabled false

File: src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        "Enabled": true,
+        "Timeout": 20000,
+        "Host": "127.0.0.1",
+        "Port": 8545,
+        "EngineHost": "127.0.0.1",
+        "EnginePort": 8551,
+        "EngineEnabledModules": "net,eth,subscribe,engine,web3,client,debug,trace,parity"
+    },
+    "Merge": {
+        "Enabled": true,
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg:29]: same question about testnet

File: src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "Blocks": {
+        "TargetBlockGasLimit": 30000000
+    },
+    "JsonRpc": {
+        "Enabled": true,
+        "Timeout": 20000,
+        "Host": "127.0.0.1",
+        "Port": 8545,
+        "EngineHost": "127.0.0.1",
+        "EnginePort": 8551,
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg:25]: Do we need EnginePort? Do we have CL ?

File: src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "Blocks": {
+        "TargetBlockGasLimit": 30000000
+    },
+    "JsonRpc": {
+        "Enabled": true,
+        "Timeout": 20000,
+        "Host": "127.0.0.1",
+        "Port": 8545,
+        "EngineHost": "127.0.0.1",
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg:24]: Probably Engine configuration is not needed

File: src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg (hunk 2 above)
Comment (shashankshampi) [reply src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg:29]: done

File: src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg (hunk 4 above)
Comment (shashankshampi) [reply src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg:24]: removed

File: src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        "NodeName": "Joc"
+    },
+    "Blocks": {
+        "TargetBlockGasLimit": 30000000
+    },
+    "JsonRpc": {
+        "Enabled": true,
+        "Timeout": 20000,
+        "Host": "127.0.0.1",
+        "Port": 8545,
```
[the remark this answers is not in the document]
Comment (shashankshampi) [reply src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg]: removed host and port

**Comment by shashankshampi** [comment]
> @shashankshampi Please add a Release Notes for this change

Done with release note and documentation about release

File: src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "Metrics": {
+        "NodeName": "Joc"
+    },
+    "Blocks": {
+        "TargetBlockGasLimit": 30000000
+    },
+    "JsonRpc": {
+        "Enabled": true,
+        "Timeout": 20000,
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Runner/configs/joc-mainnet.cfg:21]: 20s timeout is also default :)

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    "Sync": {
+        "FastSync": true,
+        "SnapSync": true,
+        "FastBlocks": true
+    },
+    "Metrics": {
+        "NodeName": "Joc"
+    },
+    "Blocks": {
+        "TargetBlockGasLimit": 30000000
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg:17]: are you sure it is correct for JOC?

Comment (kamilchodola) [reply src/Nethermind/Nethermind.Runner/configs/joc-testnet.cfg:17]: Looking at the JOC block explorer
https://explorer.japanopenchain.org/block/8453482/transactions

gas limit is 30mln

File: scripts/syncSettings.py [hunk 8]
```diff
@@ -58,6 +58,16 @@
         "url": "https://rpc.exosama.com",
         "blockReduced": 8192,
         "multiplierRequirement": 10000
+    },
+    "joc-mainnet": {
+        "url": "https://rpc-1.japanopenchain.org",
```
Comment (kamilchodola) [inline scripts/syncSettings.py:63]: Need to wait for JOC for a rpc information - these are not working so we either can't handle properly the snap/fast sync (would be very long longer-term) or we need to wait for a updated rpc.

**Review by kamilchodola — CHANGES_REQUESTED** [review]

**Review by kamilchodola — APPROVED** [review]

**Pull request merged**